### ISSUE TITLE
Explore combination of cross-entropy and dice coefficient loss

### ISSUE THREAD

**Issue by daniel-j-h** [mapbox/robosat#3]
Currently we optimize for the cross-entropy loss function in our segmentation model training.

We should try out a combination of cross-entropy and the dice coefficient, e.g.

    loss = cross-entropy - ln(dice)

or

    loss = w1 * cross-entropy - w2 * (1 - dice)  # e.g. with w1=w2=0.5

Tasks:
- [ ] Implement dice loss
- [ ] Run out experiments with a combination of cross entropy and dice loss
- [ ] Use combination of cross entropy and dice loss if it improves results

**Comment by daniel-j-h** [comment]
https://github.com/mapbox/robosat/pull/118 implements the Lovasz loss. Closing this.
